### ISSUE TITLE
Module `punycode` is deprecated since Node.js 21

### ISSUE THREAD

**Issue by cedx** [vercel/serve-handler#204]
This library depends on the [`fast-url-parser` package](https://github.com/petkaantonov/urlparser) which depends on the [`punycode` builtin module](https://nodejs.org/docs/latest/api/punycode.html)... but since Node.js 21, this module is deprecated:

```
(node:4284) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```

Issue also raised on the `fast-url-parser` repository: cf. petkaantonov/urlparser#20

**Comment by MikeMcC399** [comment]
`fast-url-parser@1.1.3` was released more than 8 years ago in Jan 2015.

[fast-url-parser > README](https://github.com/petkaantonov/urlparser/blob/master/README.md) says:

---
#API

This module has exactly the same API and semantics as the `require("url");`- module that comes with node.

See [Node.JS URL API documentation](http://nodejs.org/docs/latest/api/url.html).

---

so perhaps the `serve-handler` module could be refactored to use Node.js' `url` instead?

**Comment by MikeMcC399** [comment]
- See also https://github.com/vercel/serve/issues/802

**Comment by mamiu** [comment]
[Docusaurus](https://github.com/facebook/docusaurus) has the [same problem](https://github.com/facebook/docusaurus/issues/10107).

**Comment by slorber** [comment]
FYI I published a fork [`@docusaurus/serve-handler@6.2.0`](https://github.com/slorber/docusaurus-serve-handler) with this warning fixed.

You can install it directly, or use an alias `"serve-handler": "npm:@docusaurus/serve-handler@6.2.0",`

I do not plan to maintain it, the goal was only to fix this warning.
